backend/business-partner-agent/src/main/java/org/hyperledger/bpa/controller/api/exception/DataPersistenceExceptionHandler.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.controller.api.exception;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.server.exceptions.ExceptionHandler;
+import org.hyperledger.bpa.api.exception.DataPersistenceException;
+
+import javax.inject.Singleton;
+
+@Produces
+@Singleton
+@Requires(classes = { DataPersistenceException.class, ExceptionHandler.class })
+public class DataPersistenceExceptionHandler implements ExceptionHandler<DataPersistenceException, HttpResponse<?>> {
+    @Override
+    public HttpResponse<?> handle(HttpRequest request, DataPersistenceException exception) {
+        return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR)
```

Comment:
All uncaught exceptions will be wrapped into a 500 anyway, so no need for a special handler.